### PR TITLE
chore: adjusting pull request template

### PR DESCRIPTION
**Pull request by abernix**
This PR makes two adjustments (represented in its two commits) to the pull-request template.

1. Rename `.github/PULL_REQUEST_TEMPLATE/pull_request_template.md` to the default n ame, rather than the current location where it's required to pass the `?template=` query param to use a specific one.  (As a by-product of this, in practice the current template is not being used at all, unless someone specifies that, but I don't think anyone does.)
0. Updates our pull request template to be a bit more detailed and match that which we use on other Apollo repositories a bit more closely.  In this case, most of this is cargo cult'd from [`federation`'s repo](https://github.com/apollographql/federation/blob/main/.github/PULL_REQUEST_TEMPLATE.md).

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+<!--
+First, 🌠 thank you 🌠 for considering a contribution to Apollo!
+
```
Comment (abernix) [inline .github/pull_request_template.md:3]: I think we (I, it's my PR) should add a warning here about the state of the project and our readiness to accept contributions, one thought would be:

```suggestion
 ⚠️ Due to current roadmapping/planning and implementation work being
   worked on by the core team, this project is not yet ready to accept
   external contributions.  With the exception of perhaps _small_ bug
   fixes, please open an issue prior to opening all PRs.  We will open the
   project to contributions as soon as things are moving around a bit less
   and our roadmap is more defined.  Thanks for your understanding!
```

Comment (cecton) [reply .github/pull_request_template.md:3]: Makes sense :+1:

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Some of this information is also included in the /CONTRIBUTING.md file at the
+root of this repository.  We suggest you read it!
+
+  https://github.com/apollographql/router/blob/HEAD/CONTRIBUTING.md
```
Comment (cecton) [inline .github/pull_request_template.md:7]: I'm not sure what means HEAD here. In doubt I propose to use `main`.
```suggestion
  https://github.com/apollographql/router/blob/main/CONTRIBUTING.md
```

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        please upvote that issue by using the 👍 emote, rather than opening a
+        new issue.
+
+* 🕷 Bug fixes
+        These can be created and discussed in this repository. When fixing a bug,
+        please _try_ to add a test which verifies the fix.  If you cannot, you should
+        still submit the PR but we may still ask you (and help you!) to create a test.
+
+* 📖 Contribution guidelines
+        Follow https://github.com/apollographql/router/blob/HEAD/CONTRIBUTING.md
```
Comment (cecton) [inline .github/pull_request_template.md:29]:
```suggestion
        Follow https://github.com/apollographql/router/blob/main/CONTRIBUTING.md
```

**Review by cecton — APPROVED** [review]
Very verbose :grin: lgtm

**Review by o0Ignition0o — APPROVED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (abernix) [reply .github/pull_request_template.md:7]: I use `HEAD` to refer to the default branch which is `main` on this repository/gitremote.  I'm happy to change this to `main` if you'd like, but I opted to use this approach for referring to the default branch after the common pattern for default branch names changed to `main`.

Thoughts?

Comment (abernix) [reply .github/pull_request_template.md:7]: Put another way, if the default branch of this repository changes, assuming nobody moves `CONTRIBUTING.md`, this link will not need to be updated.  Relatedly: https://git-scm.com/docs/git-remote#Documentation/git-remote.txt-emset-headem

cc @cecton

**Comment by BrynCooke** [comment]
Perf test please

File: .github/pull_request_template.md (hunk 2 above)
Comment (cecton) [reply .github/pull_request_template.md:7]: I didn't know HEAD points to the default branch. So it's better with HEAD then!

**Pull request merged**